File: android/src/main/java/kjd/reactnative/bluetooth/BTEvent.java
package kjd.reactnative.bluetooth;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.Dynamic;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.ReadableMapKeySetIterator;
import com.facebook.react.bridge.ReadableType;
import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableMap;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public enum BTEvent {
    BLUETOOTH_ENABLED("bluetoothEnabled"),
    BLUETOOTH_DISABLED("bluetoothDisabled"),
    BLUETOOTH_CONNECTED("bluetoothConnected"),
    BLUETOOTH_DISCONNECTED("bluetoothDisconnected"),
    CONNECTION_SUCCESS("connectionSuccess"),        // Promise only
    CONNECTION_FAILED("connectionFailed"),          // Promise only
    CONNECTION_LOST("connectionLost"),
    READ("read"),
    ERROR("error");

    public final String code;
    private BTEvent(String code) {
        this.code = code;
    }

    public static WritableMap eventNames() {
        WritableMap events = Arguments.createMap();
        Arrays.stream(BTEvent.values()).forEach(e -> events.putString(e.name(), e.code));
        return events;
    }
}

File: ios/BTEvent.swift
//
//  BTEvent.swift
//  RNBluetoothClassic
//
//  Contains available Bluetooth Events used throughout the module.
//
//  Created by Ken Davidson on 2019-06-17.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

enum BTEvent : String, CaseIterable {
    case BLUETOOTH_ENABLED = "bluetoothEnabled"
    case BLUETOOTH_DISABLED = "bluetoothDisabled"
    case BLUETOOTH_CONNECTED = "bluetoothConnected"
    case BLUETOOTH_DISCONNECTED = "bluetoothDisconnected"
    case CONNECTION_SUCCESS = "connectionSuccess"        // Promise only
    case CONNECTION_FAILED = "connectionFailed"          // Promise only
    case CONNECTION_LOST = "connectionLost"
    case READ = "read"
    case ERROR = "error"
    
    static func eventNamesDictionary() -> NSDictionary {
        let events:NSDictionary = NSDictionary()
        
        for event in BTEvent.allCases {
            events.setValue(event.self, forKey: event.rawValue)
        }
        
        return events
    }
    
    static func eventNamesArray() -> [String] {
        var events:[String] = [String]()
        
        for event in BTEvent.allCases {
            events.append(event.rawValue)
        }
        
        return events;
    }
}
